### PR TITLE
Fix Bug when adding/removing layer with no cache

### PR DESCRIPTION
**Pull request by cmortazavi**
https://github.com/openlayers/openlayers/issues/6962

### CODE REVIEW

File: src/ol/renderer/layer.js [hunk 1]
```diff
@@ -152,8 +152,10 @@ ol.renderer.Layer.prototype.scheduleExpireCache = function(frameState, tileSourc
      */
     var postRenderFunction = function(tileSource, map, frameState) {
       var tileSourceKey = ol.getUid(tileSource).toString();
-      tileSource.expireCache(frameState.viewState.projection,
-          frameState.usedTiles[tileSourceKey]);
+      if (frameState.usedTiles !== null && frameState.usedTiles[tileSourceKey] !== null) {
```
Comment (ahocevar) [inline src/ol/renderer/layer.js]: `frameState.usedTiles` should never be null. And I think that `frameState.usedTiles[tileSourceKey]` would rather be `undefined` and not `null`. Does it still work if you change the `if` condition to `if (tileSourceKey in frameState.usedTiles)`?

**Comment by cmortazavi** [comment]
Changes made

**Comment by ahocevar** [comment]
Thanks @cmortazavi. We'll need a [CLA](https://github.com/openlayers/cla) from you so we can merge this.

**Comment by cmortazavi** [comment]
What is a CLA and how can I give you a CLA?

**Comment by cmortazavi** [comment]
I think I figured out how to do this.  Does everything look correct?

**Comment by ahocevar** [comment]
Thanks @cmortazavi, everything looks correct.

**Pull request merged**